libs/designer/src/lib/ui/panel/nodeDetailsPanel/tabs/mockResultsTab/mockResultsTab.tsx

```diff
@@ -37,24 +43,86 @@ export const MockResultsTab = () => {
   const mocks = useMocksByOperation(nodeName);
   const mocksValidationErrors = useMocksValidationErrors();
 
+  const [errorMessage, setErrorMessage] = useState<string>(mocks.errorMessage || '');
+  const [errorCode, setErrorCode] = useState<string>(mocks.errorCode || '');
+
   const filteredOutputs: OutputInfo[] = getFilteredOutputs(rawOutputs.outputs, nodeType);
+
+  useEffect(() => {
+    console.log('Component rendered with mocks:', mocks);
+  }, [mocks]);
+
   const onMockUpdate = useCallback(
     (id: string, newState: ChangeState, type: string) => {
       const { value, viewModel } = newState;
       const propertiesToUpdate = { value };
       if (!isNullOrUndefined(viewModel) && viewModel.arrayType === ArrayType.COMPLEX) {
         propertiesToUpdate.value = viewModel.uncastedValue;
       }
-      dispatch(updateMock({ operationName: nodeName, outputs: propertiesToUpdate.value ?? [], outputId: id, completed: true, type: type }));
+
+      console.log('Mock Update Triggered:', {
+        nodeName,
+        actionResult: mocks.actionResult,
+        id,
+        value: propertiesToUpdate.value,
+        type,
+      });
+
+      if (mocks.actionResult === ActionResults.FAILED) {
+        console.log('Dispatching updateMockFailure');
```

Comment:
or my own information, do we persist console logs anywhere that we can query for telemetry?